src/alert/__tests__/runtime-content.test.tsx

```diff
@@ -0,0 +1,305 @@
+  const plugin: AlertFlashContentConfig = {
+    id: 'test-content',
+    runReplacer(context, registerReplacement) {
+      registerReplacement('header', 'remove');
+      return {
+        update: () => {},
+        unmount: () => {},
+      };
+    },
+  };
+  awsuiPlugins.alertContent.registerContentReplacer(plugin);
+  const { container } = render(<Alert header="Initial header content" />);
+  const alertWrapper = new AlertWrapper(container);
+  await delay();
+  expectAlertContent(alertWrapper, {
+    header: false,
+    headerReplaced: true,
+  });
+});
+
+describe('runReplacer arguments', () => {
+  const runReplacer = jest.fn();
+  beforeEach(() => {
+    const plugin: AlertFlashContentConfig = {
+      id: 'test-content',
+      runReplacer,
+    };
+    awsuiPlugins.alertContent.registerContentReplacer(plugin);
+  });
+  test('refs', async () => {
+    render(
+      <Alert header="Alert header" action={<Button>Action button</Button>}>
+        Alert content
+      </Alert>
+    );
+    await delay();
+    expect(runReplacer.mock.lastCall[0].headerRef.current).toHaveTextContent('Alert header');
+    expect(runReplacer.mock.lastCall[0].contentRef.current).toHaveTextContent('Alert content');
+    expect(runReplacer.mock.lastCall[0].actionsRef.current).toHaveTextContent('Action button');
+  });
+  test('type - default', async () => {
+    render(<Alert />);
+    await delay();
+    expect(runReplacer.mock.lastCall[0].type).toBe('info');
+  });
+  test('type - custom', async () => {
+    render(<Alert type="error" />);
+    await delay();
+    expect(runReplacer.mock.lastCall[0].type).toBe('error');
+  });
+});
+
+test('calls unmount callback', async () => {
+  const unmountCallback = jest.fn();
+  const plugin: AlertFlashContentConfig = {
+    id: 'test-content',
+    runReplacer() {
+      return {
+        update: () => {},
+        unmount: unmountCallback,
+      };
+    },
+  };
+  awsuiPlugins.alertContent.registerContentReplacer(plugin);
+  const { unmount } = render(<Alert>Alert content</Alert>);
+  await delay();
+  expect(unmountCallback).not.toBeCalled();
+  unmount();
+  expect(unmountCallback).toBeCalled();
+});
+
+test('calls update callback', async () => {
+  const callback = jest.fn();
+  const plugin: AlertFlashContentConfig = {
+    id: 'test-content',
+    runReplacer: jest.fn(() => {
+      return {
+        update: callback,
+        unmount: () => {},
+      };
+    }),
+  };
+  awsuiPlugins.alertContent.registerContentReplacer(plugin);
+  const { rerender } = render(<Alert>Alert content</Alert>);
+  await delay();
+  expect(callback).toBeCalledTimes(0);
+  expect(plugin.runReplacer).toBeCalledTimes(1);
+  rerender(<Alert>Alert new content</Alert>);
+  expect(callback).toBeCalledTimes(1);
+  expect(plugin.runReplacer).toBeCalledTimes(1);
+});
+
+describe('asynchronous rendering', () => {
+  test('allows asynchronous rendering of content', async () => {
```

Comment:
what is the difference between this test and 'renders runtime content when asynchronously registered'?